run/O2HitMerger.h

```diff
@@ -135,6 +135,36 @@ class O2HitMerger : public FairMQDevice
     }
   }
 
+  // maybe we should delegate this to stack (like we do for hits)
+  void consumeMCTracks(FairMQParts& data, int& index)
+  {
+    auto mctracks = o2::Base::decodeTMessage<std::vector<o2::MCTrack>*>(data, index);
+    LOG(INFO) << "GOT " << mctracks->size() << " MCTRACKS ";
+    auto br = o2::Base::getOrMakeBranch(*mOutTree, "MCTrack", &mctracks);
+    br->SetAddress(&mctracks);
+    br->Fill();
+    br->ResetAddress();
+    index++;
+  }
+
+  void consumeTrackRefs(FairMQParts& data, int& index)
+  {
+    auto trackrefs = o2::Base::decodeTMessage<std::vector<o2::TrackReference>*>(data, index);
+    LOG(INFO) << "GOT " << trackrefs->size() << " TRACKREFS ";
+    auto br = o2::Base::getOrMakeBranch(*mOutTree, "TrackRefs", &trackrefs);
+    br->SetAddress(&trackrefs);
+    br->Fill();
+    br->ResetAddress();
+    index++;
+
+    auto itrackrefs = o2::Base::decodeTMessage<o2::dataformats::MCTruthContainer<o2::TrackReference>*>(data, index);
+    br = o2::Base::getOrMakeBranch(*mOutTree, "IndexedTrackRefs", &itrackrefs);
+    br->SetAddress(&itrackrefs);
+    br->Fill();
+    br->ResetAddress();
+    index++;
+  }
+
```

Comment:
not exactly a good example of DRY as it's basically 3 times the same code (with different types). Can't the 3 call the same function ? Would be clearer.

Reply:
Certainly can and it was on my list. Will improve it.

Reply:
Done